### PR TITLE
[rust] Load model on given device

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: extensions/tokenizers/src/main/java/ai/djl/engine/rust/RustLibrary.java [hunk 1]
```diff
@@ -22,7 +22,7 @@ private RustLibrary() {}
 
     public static native boolean isCudaAvailable();
 
-    public static native long loadModel(String modelPath, int dtype);
+    public static native long loadModel(String modelPath, int dtype, String device);
```
Comment (frankfliu) [inline extensions/tokenizers/src/main/java/ai/djl/engine/rust/RustLibrary.java:25]: Why we use String, can we just use device id?

Comment (xyang16) [reply extensions/tokenizers/src/main/java/ai/djl/engine/rust/RustLibrary.java:25]: How distinguish cpu and cuda?

File: extensions/tokenizers/rust/src/models/mod.rs [hunk 2]
```diff
@@ -234,3 +234,21 @@ pub extern "system" fn Java_ai_djl_engine_rust_RustLibrary_runInference<'local>(
         }
     }
 }
+
+pub fn as_device(device: &String) -> Result<Device> {
```
Comment (frankfliu) [inline extensions/tokenizers/rust/src/models/mod.rs:238]: Can we re-use existing `as_device()`

File: extensions/tokenizers/src/main/java/ai/djl/engine/rust/RustLibrary.java (hunk 1 above)
Comment (frankfliu) [reply extensions/tokenizers/src/main/java/ai/djl/engine/rust/RustLibrary.java:25]: https://github.com/deepjavalibrary/djl/blob/master/engines/pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/jni/JniUtils.java#L1785

Comment (frankfliu) [reply extensions/tokenizers/src/main/java/ai/djl/engine/rust/RustLibrary.java:25]:
```
public static native long loadModel(String modelPath, int dtype, int deviceType, int deviceId);
```

Comment (xyang16) [reply extensions/tokenizers/src/main/java/ai/djl/engine/rust/RustLibrary.java]: Done.

File: extensions/tokenizers/rust/src/models/mod.rs (hunk 2 above)
Comment (xyang16) [reply extensions/tokenizers/rust/src/models/mod.rs]: Ok, I have modified the existing `as_device()` a little bit, please review.

File: extensions/tokenizers/rust/src/models/mod.rs [hunk 3]
```diff
@@ -167,14 +146,25 @@ fn load_model<'local>(
...
+    device_type: JString,
+    device_id: jint,
 ) -> jlong {
-    let model = load_model(&mut env, model_path, dtype);
+    let model = || {
+        let model_path: String = env
+            .get_string(&model_path)
+            .expect("Couldn't get java string!")
+            .into();
+        let dtype = as_data_type(dtype).unwrap();
```
[the remark this answers is not in the document]
Comment (xyang16) [reply extensions/tokenizers/rust/src/models/mod.rs]: Done

**Review by frankfliu — APPROVED** [review]

**Pull request merged**